deck/views.py

```diff
@@ -31,16 +31,36 @@ class BaseEventView(object):
 class ListEvents(BaseEventView, ListView):
     template_name = 'event/event_list.html'
     queryset = Event.objects.published_ones()
+    allow_empty = True
 
     def get_context_data(self, **kwargs):
         context = super(ListEvents, self).get_context_data(**kwargs)
         if self.request.user.is_authenticated():
-            event_list = Event.objects.filter(
-                models.Q(is_published=True) |
+            event_list = context['event_list'].filter(
                 models.Q(author=self.request.user))
             context.update(event_list=event_list)
         return context
 
+    def post(self, request, *args, **kwargs):
```

Comment:
Another problem I see here is that all the logic is inside a view.
I am not 100% against it (99%) but we are injecting business rules inside the framework stuff (views, HTTP, etc).

If you understand this but you are not sure if I am right, please, read some content about hexagonal architecture. Enjoy! \o/